torchvision/prototype/datasets/_builtin/usps.py

```diff
@@ -0,0 +1,61 @@
+from typing import Any, Dict, List, Tuple
+
+import numpy as np
+import torch
+from torchdata.datapipes.iter import IterDataPipe, LineReader, Mapper, Decompressor
+from torchvision.prototype.datasets.utils import Dataset, DatasetInfo, DatasetConfig, OnlineResource, HttpResource
+from torchvision.prototype.datasets.utils._internal import hint_sharding, hint_shuffling
+from torchvision.prototype.features import Image, Label
+
+
+class USPS(Dataset):
+    def _make_info(self) -> DatasetInfo:
+        return DatasetInfo(
+            "usps",
+            homepage="https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multiclass.html#usps",
+            valid_options=dict(
+                split=("train", "test"),
+            ),
+            categories=10,
+        )
+
+    _URL = "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multiclass"
+
+    _RESOURCES = {
+        "train": HttpResource(
+            f"{_URL}/usps.bz2", sha256="3771e9dd6ba685185f89867b6e249233dd74652389f263963b3b741e994b034f"
+        ),
+        "test": HttpResource(
+            f"{_URL}/usps.t.bz2", sha256="a9c0164e797d60142a50604917f0baa604f326e9a689698763793fa5d12ffc4e"
+        ),
+    }
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+        return [USPS._RESOURCES[config.split]]
+
+    def _prepare_sample(self, data: Tuple[torch.Tensor, torch.Tensor]) -> Dict[str, Any]:
+        _filename, line = data
```

Comment:
With the suggestion above, this simplifies to

```suggestion
    def _prepare_sample(self, line: str) -> Dict[str, Any]:
```